File: 04 - ORIENTAÇÃO A OBJETOS/nodeJS - JavaScript_/Exercicio 7/Com Pacotes/Aluno.js
var Pessoa = require('./Pessoa')

module.exports = class Aluno extends Pessoa {
    constructor(nome, endereco, idade, curso, listaDisc){
        super(nome, endereco, idade, listaDisc)
        this.curso = curso
    }

    printDescricao(){
        console.log(`Aluno: ${this.nome}`)
        console.log(`Idade: ${this.idade}`)
        console.log(`Curso: ${this.curso}`)
        const lista = this.listaDisc
        for(let i=0; i<lista.length; i++)
            console.log(`Nome: ${lista[i].nomeDisc} - Carga Horária: ${lista[i].cargaHoraria}`)
    }
}

File: 04 - ORIENTAÇÃO A OBJETOS/Java/Exercicio 8/Aluno.java

import java.util.ArrayList;

public class Aluno extends Pessoa {
    String curso;
    
    public Aluno(String nome, String endereco, int idade, String curso, ArrayList<Disciplina> listaDisc) {
        super(nome, endereco, idade, listaDisc);
        this.curso = curso;
    }
    
    String getCurso(){
        return this.curso;
    }

    @Override
    void printDescricao() {
        System.out.println("Aluno: "+super.getNome());
        System.out.println("Endereço: "+super.getEndereco());
        System.out.println("Idade: "+super.getIdade());
        System.out.println("Curso: "+this.curso);
        System.out.println("Disciplina(s) cursada(s): ");
        ArrayList<Disciplina> listaDisc = super.getListaDisc();
        for(Disciplina disc : listaDisc){
            System.out.println("Nome: "+disc.getNomeDisc()+" - Carga Horária: "+disc.getCargaHoraria());
        }
    }
    
}
